library/src/image.cxx

```diff
@@ -19,48 +22,35 @@ namespace f3d
 class image::internals
 {
 public:
-  std::vector<unsigned char> Buffer;
-  unsigned int Width = 0;
-  unsigned int Height = 0;
-  unsigned int Channels = 0;
-
-  void UpdateBufferSize() { this->Buffer.resize(this->Width * this->Height * this->Channels); }
-
-  vtkSmartPointer<vtkImageImport> GetVTKImporter() const
-  {
-    assert(this->Buffer.size() == this->Width * this->Height * this->Channels);
-
-    vtkNew<vtkImageImport> importer;
-    importer->CopyImportVoidPointer(
-      const_cast<unsigned char*>(this->Buffer.data()), static_cast<vtkIdType>(this->Buffer.size()));
-    importer->SetNumberOfScalarComponents(this->Channels);
-    importer->SetDataScalarTypeToUnsignedChar();
-    importer->SetWholeExtent(0, this->Width - 1, 0, this->Height - 1, 0, 0);
-    importer->SetDataExtentToWholeExtent();
-    return importer;
-  }
-
-  void SetFromVTK(vtkAlgorithm* alg)
-  {
-    vtkNew<vtkImageExport> exporter;
-    exporter->SetInputConnection(alg->GetOutputPort());
-    exporter->ImageLowerLeftOn();
-
-    int* dims = exporter->GetDataDimensions();
-    this->Width = dims[0];
-    this->Height = dims[1];
-    this->Channels = exporter->GetDataNumberOfScalarComponents();
-    this->UpdateBufferSize();
-
-    exporter->Export(this->Buffer.data());
-  }
+  vtkSmartPointer<vtkImageData> Image;
```

Comment:
could you explain in a few words why changing the implementation logic is needed ?

Reply:
I was managing the buffer, resolution, and number of channels. Adding the scalar type was too much work and I realized that I could just rely on `vtkImageData` which already supports everything we need.

Reply:
right, VTK rules after all :)